src/lib/es5.d.ts

```diff
@@ -210,6 +210,8 @@ interface ObjectConstructor {
       * Returns the names of the enumerable properties and methods of an object.
       * @param o Object that contains the properties and methods. This can be an object that you created or an existing Document Object Model (DOM) object.
       */
+    keys<T>(o: Array<T>): string[];
+    keys<T extends { [key: string]: any }>(o: T): (keyof T & string)[];

```

Comment:
@mhegazy No, sometimes keyof T is `string | number`. That's part of the problem. See this test case: https://github.com/Microsoft/TypeScript/pull/12253/files#diff-3ed8d911aae864ffc1d88e62bcb8dc47R15